@here/harp-map-controls/lib/CameraKeyTrackAnimation.ts

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates, GeoCoordLike } from "@here/harp-geoutils";
+import { LookAtParams, MapView, MapViewEventNames, MapViewUtils } from "@here/harp-mapview";
+import { LoggerManager } from "@here/harp-utils";
+import THREE = require("three");
+
+const logger = LoggerManager.instance.create("CameraKeyTrackAnimation");
+
+export interface ControlPointOptions extends Partial<LookAtParams> {
+    /**
+     * @override
+     */
+    target: GeoCoordinates | GeoCoordLike;
+    /**
+     * A timestamp for the [[ControlPoint]], describes when during the animation the
+     * [[ControlPoint]] should be reached, in seconds
+     */
+    timestamp: number;
+    /**
+     * @override
+     */
+    tilt?: number;
+    /**
+     * @override
+     */
+    heading?: number;
+    /**
+     * @override
+     */
+    distance?: number;
+    /**
+     * A name for the ControlPoint
+     */
+    name?: string;
+}
+/**
+ * A ControlPoint used to defined the Track of the [[CameraKeyTrackAnimation]]
+ */
+export class ControlPoint
+    implements Pick<LookAtParams, "target" | "tilt" | "heading" | "distance"> {
+    /**
+     * The time, when this CameraPose should be reached in the animation flow, in seconds.
+     */
+    timestamp: number = 0;
+    target: GeoCoordinates;
+    name?: string;
+    tilt: number;
+    heading: number;
+    distance: number;
+
+    constructor(options: ControlPointOptions) {
+        this.timestamp = options.timestamp;
+        this.target = options.target
+            ? GeoCoordinates.fromObject(options.target)
+            : new GeoCoordinates(0, 0);
+        this.tilt = options.tilt ?? 0;
+        this.heading = options.heading ?? 0;
+        this.distance = options.distance ?? 0;
+        this.name = options.name ?? Date.now().toString();
+    }
+}
+
+export interface CameraKeyTrackAnimationOptions {
+    /**
+     * The [[ControlPoint]]s the animation should pass
+     */
+    controlPoints: ControlPoint[];
+    /**
+     * If ´true` the animation restarts after finishing, therefore repeats. @default false.
+     */
+    repeat?: boolean;
+    /**
+     * An optional name for the animation, @default ""
+     */
+    name?: string;
+    /**
+     * The [[THREE.InterpolationModes]]  for the animation.
+     * @default THREE.InterpolateSmooth
+     */
+    interpolation?: THREE.InterpolationModes;
+    /**
+     * See [[THREE.AnimationActionLoopStyles]]
+     * @default THREE.LoopOnce
+     */
+    loop?: THREE.AnimationActionLoopStyles; //types seem outdate too, should be THREE.LoopModes
+    /**
+     * If `true`the Animation will interpolate rotations only clockwise.
+     * @default true
+     */
+    rotateOnlyClockwise?: boolean;
+    /**
+     * Defines how often the Animation should be repeated.
+     * @default 1
+     */
+    repetitions?: number;
```

Comment:
How does this play together with `loop`. What happens for example if loop is `LoopOnce` and repetitions is 2?

Reply:
Oh, good catch... the repetitions and loop actually go together in the THREE.AnimationAction         this.m_animationAction.setLoop(loop, repetitions);

But the "repeat" was a leftover, removed it

Reply:
I added these parameters to the example, so it can be tried out how it works together